### DISCUSSION TITLE
Autofix capability of linters

### DISCUSSION THREAD

**Discussion by sahidvelji** [golangci/golangci-lint#4485] [Q&A] [answered]
Some of the listed linters indicate autofix capability: https://golangci-lint.run/usage/linters
However, linters like `perfsprint`, `wsl`, `unconvert`, `testifylint`, etc. also have autofix capabilities. Can these also be added to golangci-lint, or am I missing something?

**Comment by bombsimon** [comment, marked as answer]
This is related to #1779. The linters you're mentioning supports `SuggestedFixes` but `golangci-lint` has its own fix system that was built before it was a part of the `analysis` package.

The linked issue is a tracking issue to support this. There's currently no automatic way to support this without adding a separate fixer for `golangci-lint` so it would be better to add supprot for `SuggestedFixes`.

One reason it was postponed was because it wasn't stable and only a [proposed api](https://github.com/golang/tools/blob/master/go/analysis/doc/suggested_fixes.md#proposed-suggested-fix-api) ([old comment](https://github.com/golang/tools/blob/b4639ccb830b1c9602f1c96eba624bbbe20650ea/go/analysis/diagnostic.go#L23-L28)). It has since been removed and is now stable ([current comment](https://github.com/golang/tools/blob/ca94c9663f9ac326811346916fe3444e95c34e1f/go/analysis/diagnostic.go#L34-L37)).
